pytorch_lightning/plugins/training_type/ddp.py

```diff
@@ -155,19 +156,24 @@ def _call_children_scripts(self):
         os.environ["NODE_RANK"] = str(self.cluster_environment.node_rank())
         os.environ["LOCAL_RANK"] = str(self.cluster_environment.local_rank())
 
-        # when user is using hydra find the absolute path
-        path_lib = os.path.abspath if not _HYDRA_AVAILABLE else to_absolute_path
-
-        # pull out the commands used to run the script and resolve the abs file path
-        command = sys.argv
-        try:
-            full_path = path_lib(command[0])
-        except Exception:
-            full_path = os.path.abspath(command[0])
-
-        command[0] = full_path
-        # use the same python interpreter and actually running
-        command = [sys.executable] + command
+        # Check if the current calling command looked like `python a/b/c.py` or `python -m a.b.c`
+        # See https://docs.python.org/3/reference/import.html#main-spec
+        if __main__.__spec__ is None:  # pragma: no-cover . Script called as `python a/b/c.py`
```

Comment:
```suggestion
        if __main__.__spec__ is None:  # pragma: no-cover
            # script called as `python a/b/c.py`
```

Reply:
@nisheethlahoti the comment was good, I just meant to say to move it below the if, not to interfere with the pragma

Reply:
The way codecov works, it skips any line that has "pragma: no-cover" as a substring. So the existing comment doesn't interfere with the pragma either. Anyway, moving.